platform/jvm/capture/src/main/kotlin/io/bitdrift/capture/LoggerImpl.kt

```diff
@@ -265,6 +266,16 @@ internal class LoggerImpl(
         appExitSaveCurrentSessionId()
     }
 
+    override fun shutdown() {
+        runCatching {
+            appExitLogger.uninstallAppExitLogger()
+            eventsListenerTarget.stop()
+            // TODO(murki): Should we run this? in a bg thread?
+            CaptureJniLibrary.flush(this.loggerId, true)
+            CaptureJniLibrary.destroyLogger(this.loggerId)
```

Comment:
Need to confirm that calling `destroyLogger()` will destroy the threads used by `resource_utilization_target` and `session_replay_target` and `events_listener_target`

Reply:
It will do so asynchronously, calling destroyLogger will trigger the shutdown notification to be sent to active tasks which should have them wind down as they are able to respond to the signal, then once all the primary tasks shut down we will turn down the runtime executor. 

All of those tasks seem to subscribe to the shutdown signal so I believe those tasks should wind down in response to calling shutdown